device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/qos.json.j2

```diff
@@ -100,109 +103,207 @@
+            "type": "DWRR",
+            "weight": "22"
         },
-        "{{ port }}|2": {
-            "scheduler": "scheduler.18",
-            "wred_profile": "AZURE_LOSSY_Q2"
+        "scheduler_q5_downlink": {
+            "type": "DWRR",
+            "weight": "22"
         },
-        "{{ port }}|3": {
-            "scheduler": "scheduler.22",
-            "wred_profile": "AZURE_LOSSY_Q3"
+        "scheduler_q6_downlink": {
+            "type": "DWRR",
+            "weight": "4"
         },
-        "{{ port }}|4": {
-            "scheduler": "scheduler.22"
+        "scheduler_q0_uplink": {
+            "type": "DWRR",
+            "weight": "4"
         },
-        "{{ port }}|5": {
-            "scheduler": "scheduler.22"
+        "scheduler_q1_uplink": {
+            "type": "DWRR",
+            "weight": "8"
         },
-        "{{ port }}|6": {
-            "scheduler": "scheduler.4"
+        "scheduler_q2_uplink": {
+            "type": "DWRR",
+            "weight": "18"
+        },
+        "scheduler_q3_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q4_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q5_uplink": {
+            "type": "DWRR",
+            "weight": "22"
+        },
+        "scheduler_q6_uplink": {
+            "type": "DWRR",
+            "weight": "4"
         }
+    },
+{%- endmacro -%}
+
+{%- macro generate_direction_based_queue_per_sku(port, direction) -%}
+    "{{ port }}|0": {
+        "scheduler": "scheduler_q0_{{ direction }}"
+    },
+    "{{ port }}|1": {
+        "scheduler": "scheduler_q1_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q1"
+    },
+    "{{ port }}|2": {
+        "scheduler": "scheduler_q2_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q2"
+    },
+    "{{ port }}|3": {
+        "scheduler": "scheduler_q3_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q3"
+    },
+    "{{ port }}|4": {
+        "scheduler": "scheduler_q4_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q4"
+    },
+    "{{ port }}|5": {
+        "scheduler": "scheduler_q5_{{ direction }}",
+        "wred_profile": "AZURE_LOSSY_{{ direction.upper() }}_Q5"
+    },
+    "{{ port }}|6": {
+        "scheduler": "scheduler_q6_{{ direction }}"
+    }
 {%- endmacro -%}
 
 {%- macro generate_global_dscp_to_tc_map() %}
 {# This is an empty macro since the global DSCP_TO_TC map is not required #}
 {%- endmacro %}
 
+{# downlink Q use t0, uplink Q use t1 #}
 {%- macro generate_wred_profiles() %}
-    {%- set max_threshold = "282624" %}
+    {# Default threshold values for both ToR and Leaf routers #}
+    {%- set t0_max_threshold = "282624" %}  {# Default t0 threshold for downlink ports #}
+    {%- set t1_max_threshold = "239616" %}  {# Default t1 threshold for uplink ports #}
```

Comment:
what's difference among L201-202, L210-211, L216-217?
I believe the correct logic should be
1. To define the max threshold according to t0/t1
2. To reference the max threshold in each profile

Reply:
Done

Reply:
Now fixed, thanks